create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java

```diff
@@ -278,89 +252,21 @@ private boolean isScopusIdentifier(ItemidTp itemIdTp) {
 
     private AdditionalIdentifier toAdditionalIdentifier(ItemidTp itemIdTp) {
         return new AdditionalIdentifier(ScopusConstants.ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME,
-                itemIdTp.getValue());
+                                        itemIdTp.getValue());
     }
 
     private PublicationContext getPublicationContext() {
         if (isJournal()) {
-            return attempt(() -> createPeriodical())
-                    .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+            return new JournalCreator(metadataService, docTp).createJournal();
```

Comment:
moved the Journal context generation in a separate class for readability

Reply:
Acctually I think we could even go further end make it a PublicationContextCreator. A lot of tasks/method will probably repeat itslef if we are dealing with books and what ever publication.

Reply:
I agree, PublicationContextCreator will be an abstract class inherited by JournalCreator, BookCreator etc.